diff --git a/include/configs/rv1106_common.h b/include/configs/rv1106_common.h
index dc927a8..7922f7b 100644
--- a/include/configs/rv1106_common.h
+++ b/include/configs/rv1106_common.h
@@ -35,7 +35,7 @@
 /* SPL */
 #define CONFIG_SPL_FRAMEWORK
 #define CONFIG_SPL_TEXT_BASE		0x00000000
-#define CONFIG_SPL_MAX_SIZE		0x28000
+#define CONFIG_SPL_MAX_SIZE		0x70000
 #define CONFIG_SPL_BSS_START_ADDR	0x001fe000
 #define CONFIG_SPL_BSS_MAX_SIZE		0x20000
 #define CONFIG_SPL_STACK		0x001fe000
@@ -89,14 +89,11 @@
 
 #undef RKIMG_BOOTCOMMAND
 #ifdef CONFIG_FIT_SIGNATURE
-#define RKIMG_BOOTCOMMAND		\
-	"boot_fit;"
 #else
-#define RKIMG_BOOTCOMMAND		\
-	"boot_fit;"			\
-	"boot_android ${devtype} ${devnum};"
 #endif
 
+#define RKIMG_BOOTCOMMAND "fatload mmc 1:5 0x8C00000 rv1106-custom.dtb;fatload mmc 1:5 0x8000800 zImage;bootz 0x8000800 - 0x8C00000"
+
 /* Update define for tiny image */
 #ifdef CONFIG_ROCKCHIP_IMAGE_TINY
 #undef RKIMG_BOOTCOMMAND
@@ -108,7 +105,6 @@
 #undef CONFIG_GZIP
 /* TODO: #define CONFIG_LIB_HW_RAND */
 
-#define RKIMG_BOOTCOMMAND		"boot_fit;"
 #define CONFIG_EXTRA_ENV_SETTINGS	ENV_MEM_LAYOUT_SETTINGS
 #endif
 #endif	/* !CONFIG_SPL_BUILD */
