diff --git a/drivers/pinctrl/rockchip/pinctrl-rockchip-core.c b/drivers/pinctrl/rockchip/pinctrl-rockchip-core.c
index c3fb549899..d116962f77 100644
--- a/drivers/pinctrl/rockchip/pinctrl-rockchip-core.c
+++ b/drivers/pinctrl/rockchip/pinctrl-rockchip-core.c
@@ -42,7 +42,7 @@ void rockchip_get_recalced_mux(struct rockchip_pin_bank *bank, int pin,
 {
 	struct rockchip_pinctrl_priv *priv = bank->priv;
 	struct rockchip_pin_ctrl *ctrl = priv->ctrl;
-	struct rockchip_mux_recalced_data *data;
+	struct rockchip_mux_recalced_data *data = NULL;
 	int i;
 
 	for (i = 0; i < ctrl->niomux_recalced; i++) {
@@ -55,6 +55,9 @@ void rockchip_get_recalced_mux(struct rockchip_pin_bank *bank, int pin,
 	if (i >= ctrl->niomux_recalced)
 		return;
 
+	if (data == NULL)
+		return;
+
 	*reg = data->reg;
 	*mask = data->mask;
 	*bit = data->bit;
@@ -66,7 +69,7 @@ rockchip_get_mux_route(struct rockchip_pin_bank *bank, int pin,
 {
 	struct rockchip_pinctrl_priv *priv = bank->priv;
 	struct rockchip_pin_ctrl *ctrl = priv->ctrl;
-	struct rockchip_mux_route_data *data;
+	struct rockchip_mux_route_data *data = NULL;
 	int i;
 
 	for (i = 0; i < ctrl->niomux_routes; i++) {
@@ -79,6 +82,9 @@ rockchip_get_mux_route(struct rockchip_pin_bank *bank, int pin,
 	if (i >= ctrl->niomux_routes)
 		return ROUTE_TYPE_INVALID;
 
+	if (data == NULL)
+		return ROUTE_TYPE_INVALID;
+
 	*reg = data->route_offset;
 	*value = data->route_val;
 
